docs/reference/index-modules/slowlog.asciidoc

```diff
@@ -58,29 +58,85 @@ The search slow log file is configured in the `log4j2.properties` file.
+
+[source,console]
+--------------------------------------------------
+PUT /my-index-000001/_settings
+{
+  "index.search.slowlog.include.user": true
+}
+--------------------------------------------------
+// TEST[setup:my_index]
+
+This will result in user information being included in the slow log.
 
 [source,js]
 ---------------------------
 {
-  "type": "index_search_slowlog",
-  "timestamp": "2030-08-30T11:59:37,786+02:00",
-  "level": "WARN",
-  "component": "i.s.s.query",
-  "cluster.name": "distribution_run",
-  "node.name": "node-0",
-  "message": "[index6][0]",
-  "took": "78.4micros",
-  "took_millis": "0",
-  "total_hits": "0 hits",
-  "stats": "[]",
-  "search_type": "QUERY_THEN_FETCH",
-  "total_shards": "1",
-  "source": "{\"query\":{\"match_all\":{\"boost\":1.0}}}",
-  "id": "MY_USER_ID",
-  "cluster.uuid": "Aq-c-PAeQiK3tfBYtig9Bw",
-  "node.id": "D7fUYfnfTLa2D7y-xw6tZg"
+  "@timestamp": "2024-02-21T12:42:37.255Z",
+  "log.level": "WARN",
+  "auth.type": "REALM",
+  "elasticsearch.slowlog.id": null,
+  "elasticsearch.slowlog.message": "[index6][0]",
+  "elasticsearch.slowlog.search_type": "QUERY_THEN_FETCH",
+  "elasticsearch.slowlog.source": "{\"query\":{\"match_all\":{\"boost\":1.0}}}",
+  "elasticsearch.slowlog.stats": "[]",
+  "elasticsearch.slowlog.took": "747.3micros",
+  "elasticsearch.slowlog.took_millis": 0,
+  "elasticsearch.slowlog.total_hits": "1 hits",
+  "elasticsearch.slowlog.total_shards": 1,
+  "user.effective.name": "elastic",
+  "user.effective.realm": "reserved",
+  "user.name": "elastic",
+  "user.realm": "reserved",
+  "ecs.version": "1.2.0",
+  "service.name": "ES_ECS",
+  "event.dataset": "elasticsearch.index_search_slowlog",
+  "process.thread.name": "elasticsearch[runTask-0][search][T#5]",
+  "log.logger": "index.search.slowlog.query",
+  "elasticsearch.cluster.uuid": "Ui23kfF1SHKJwu_hI1iPPQ",
+  "elasticsearch.node.id": "JK-jn-XpQ3OsDUsq5ZtfGg",
+  "elasticsearch.node.name": "node-0",
+  "elasticsearch.cluster.name": "distribution_run"
+}
+
+---------------------------
+// NOTCONSOLE
+
+If a call was initiated with an `X-Opaque-ID` header, then the ID is also included
+in Search Slow logs as an additional **elasticsearch.slowlog.id** field
+
+[source,js]
+---------------------------
+{
+  "@timestamp": "2024-02-21T12:42:37.255Z",
+  "log.level": "WARN",
+  "auth.type": "REALM",
+  "elasticsearch.slowlog.id": "MY_USER_ID",
+  "elasticsearch.slowlog.message": "[index6][0]",
+  "elasticsearch.slowlog.search_type": "QUERY_THEN_FETCH",
+  "elasticsearch.slowlog.source": "{\"query\":{\"match_all\":{\"boost\":1.0}}}",
+  "elasticsearch.slowlog.stats": "[]",
+  "elasticsearch.slowlog.took": "747.3micros",
+  "elasticsearch.slowlog.took_millis": 0,
+  "elasticsearch.slowlog.total_hits": "1 hits",
+  "elasticsearch.slowlog.total_shards": 1,
+  "user.effective.name": "elastic",
+  "user.effective.realm": "reserved",
+  "user.name": "elastic",
+  "user.realm": "reserved",
```

Comment:
this is the duplication i referenced in my other comment. I think we should avoid emiting duplicates and most people won't understand the difference. I think (but could be convinced otherwise) we should keep it simple and always use user.effective.name and document that it may not have been the user that initially authenticated. This also matches the audit trail: https://www.elastic.co/guide/en/elasticsearch/reference/current/audit-event-types.html  This is also where the x-opaque-id can help to track down who called it and where the audit trail can help provide who authenticated (user.run_by.name).

Reply:
+1 to conditionally showing the second value iff it is not a duplicate. 

However, can we align the names and values used here with the audit log ?
`user.name` = effective name
`user.run_by.name` = authenticating name

Also, unless it creates a weird dependency we should probably use the final String values from `org.elasticsearch.xpack.security.audit.logfile` to help enforce this matches what you see in the audit log.

Reply:
I'm not sure which is the right path forward here. Based on you experience, is it more important to align with the audit log or with ECS (ECS uses [effective_name](https://www.elastic.co/guide/en/ecs/current/ecs-user.html#ecs-user-nestings) and the audit log `run_by` to nest the user)? 

I agree with using the constants if possible, good call out!

Reply:
> , is it more important to align with the audit log or with ECS 

ECS. Ideally the audit log would align with ECS as well, but I believe it predates ECS.